cmd/podman/main_local.go

```diff
@@ -139,6 +142,32 @@ func setupRootless(cmd *cobra.Command, args []string) error {
 		Remote:      remoteclient,
 	}
 
+	runtime, err := libpodruntime.GetRuntime(getContext(), &podmanCmd)
+	if err != nil {
+		return errors.Wrapf(err, "could not get runtime")
+	}
+	defer runtime.DeferredShutdown(false)
+
+	// do it only after podman has already re-execed and running with uid==0.
+	if os.Geteuid() == 0 {
+		conf, err := runtime.GetConfig()
+		if err != nil {
+			return err
+		}
+		unitName := fmt.Sprintf("podman-%d.scope", os.Getpid())
+		if err := utils.RunUnderSystemdScope(os.Getpid(), "user.slice", unitName); err != nil {
```

Comment:
Should this be unconditional? I feel like we only want it with systemd CGroups.

Reply:
we also want this when using cgroupfs.  It solves the problem of running systemd containers on cgroups v1 with: systemd-run --scope --user podman ...

If it fails though, we just print a debug message